### ISSUE TITLE
Fix angle brackets vs. CKEditor

### ISSUE THREAD

**Issue by olafgrabienski** [backdrop-contrib/footnotes#14]
Footnotes is supposed to allow the use of `[fn]square brackets[/fn]` and `<fn>angle brackets</fn>` shortcodes.

On my testing site, I can use square brackets without any issue. When I use angle brackets, the shortcode isn't be displayed, instead of a footnote I see the shortcode on the page. When I edit the page and switch the editor to *Source code*, I see that the angle bracket short code is converted to something like `&lt;fn&gt;My footnote text.&lt;/fn&gt;`.

First I thought it was an issue with the ordering of filters in the text format configuration. From Drupal I know, that Footnotes was designed so that it should run before HTML filter, and that's the case on my testing site. I've no idea at the moment why the angle brackets don't work.

**Update:** The angle brackets syntax works generally, but it does *not* work when CKEditor is enabled in the text format configuration. Using CKEditor is a very common use case though, so we should try to fix the issue anyway.

**Comment by olafgrabienski** [comment]
Personally, I prefer to use square brackets, and I think they're generally less problematic because they differ clearly from HTML syntax.

We could consider to drop support for angle brackets but before we should check at least which type of brackets is used by the CKEditor button. We should also consider if dropping support for angle brackets will prevent people to use the module who upgrade from Drupal 7. (Will file eventually a separate issue)

**Comment by olafgrabienski** [comment]
> We could consider to drop support for angle brackets but before we should check at least which type of brackets is used by the CKEditor button.

I've had a look at it: Footnotes inserted by the CKEditor button use square brackets. So, the CKEditor integration wouldn't prevent us from dropping support for angle brackets.

I'm still not sure though if we should keep angle brackets for the sake of Drupal 7 compability. What do others think about?

**Comment by herbdool** [comment]
Some people might want old content to still work if they used angle brackets, who knows how many people, probably few.

It might be ckeditor rewriting before it's even saved. Maybe test where ckeditor is disabled.

**Comment by olafgrabienski** [comment]
@herbdool Thanks for your feedback! Yes, hard to say how many people would want to use angle brackets. However, in Drupal not less than 2,651 sites report using the module, and some of them might be interested in Backdrop. So, for the time being let's **not** drop angle brackets support.

**Comment by olafgrabienski** [comment]
> It might be ckeditor rewriting before it's even saved. Maybe test where ckeditor is disabled.

Thanks for the hint! Indeed, when I disable CKEditor temporarily, the angle brackets syntax works. (Will add it to the issue description.)

**Comment by olafgrabienski** [comment]
Unfortunately, I don't know how to debug the issue further. Any help welcome!

**Comment by herbdool** [comment]
Maybe just leave it and remove help text which references angle brackets.

**Comment by olafgrabienski** [comment]
Good idea! That would allow me to make a release, and when ported Drupal sites need angle brackets, we could try to fix it again with the help of them.

Removing the `release blocker` label.

Note to myself: have a look at places where the angle brackets are referenced

**Comment by herbdool** [comment]
Plus if you leave in the capability to interpret the angle brackets then old content will still work (so long as it's not edited again).
